src/NoSqlProvider.ts

```diff
@@ -46,6 +47,7 @@ export interface DbIndex {
     countOnly(key: any|any[]): SyncTasks.Promise<number>;
     countRange(keyLowRange: any|any[], keyHighRange: any|any[], lowRangeExclusive?: boolean, highRangeExclusive?: boolean)
         : SyncTasks.Promise<number>;
+    fullTextSearch<T>(searchPhrase: string): SyncTasks.Promise<T[]>;
```

Comment:
There shouldn't be anything past an extra if check here and there.  If you notice anything substantial we can think about it, but the intent was not to damage the normal hot paths.